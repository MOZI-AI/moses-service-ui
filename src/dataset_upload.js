import React from "react";
import { Upload, Icon, Alert, Button, Divider, message } from "antd";

export class DatasetUpload extends React.Component {
  componentDidMount() {
  }

  render() {
    const props = {
      name: "dataset",
      multiple: false,
      beforeUpload: file => {
        console.log('file ', file);
        if (file.type === 'text/csv') {
          this.props.handleFileUpload(file);
        } else {
          message.error('File must be in .csv format.');
        }

        return false;
      }
    };

    return (
      this.props.show && <div className="datasetUploadWrapper">
        <Upload.Dragger {...props} fileList={[]}>
          <p className="ant-upload-drag-icon">
            <Icon type="upload" />
          </p>
          <p className="ant-upload-text">
            {!this.props.uploadedFile ? "Click or drag dataset file to this area to upload" : "You may upload another dataset file to replace the current one."}
          </p>
          <p className="ant-upload-hint">
            Please make sure the dataset file is in .csv format.
          </p>
        </Upload.Dragger>
        {this.props.uploadedFile && (
          <Alert className="fileDetails"
            style={{ borderRadius: 0, borderTop: "none" }}
            message={this.props.uploadedFile.name}
            type="info"
          />
        )}
        <Divider dashed />
        <Button
          type="primary"
          className="nextButton"
          disabled={!this.props.uploadedFile}
          onClick={e => this.props.next()}
        >
          Next <Icon type="right" />
        </Button>
      </div>
    );
  }
}
