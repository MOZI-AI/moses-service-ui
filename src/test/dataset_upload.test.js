import React from 'react';
import { shallow } from 'enzyme';
import { DatasetUpload } from '../dataset_upload';
import renderer from 'react-test-renderer';

describe('<DatasetUpload />', () => {
  it('rendered correctly', () => {
    const tree = renderer.create(<DatasetUpload show="true" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('starts without an uploaded file', () => {
    const wrapper = shallow(<DatasetUpload show="true" />);
    expect(wrapper.find('div#fileDetails').exists()).toBeFalsy();
  });
});
