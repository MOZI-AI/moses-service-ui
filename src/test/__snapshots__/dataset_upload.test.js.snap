// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatasetUpload /> rendered correctly 1`] = `
<div
  className="datasetUploadWrapper"
>
  <span
    className=""
  >
    <div
      className="ant-upload ant-upload-drag"
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
    >
      <span
        className="ant-upload ant-upload-btn"
        onClick={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "height": undefined,
          }
        }
        tabIndex="0"
      >
        <input
          accept=""
          directory={null}
          multiple={false}
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
          webkitdirectory={null}
        />
        <div
          className="ant-upload-drag-container"
        >
          <p
            className="ant-upload-drag-icon"
          >
            <i
              className="anticon anticon-upload"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="upload"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 0 0-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </i>
          </p>
          <p
            className="ant-upload-text"
          >
            Click or drag dataset file to this area to upload
          </p>
          <p
            className="ant-upload-hint"
          >
            Please make sure the dataset file is in .csv format.
          </p>
        </div>
      </span>
    </div>
    <div
      className="ant-upload-list ant-upload-list-text"
    />
  </span>
  <div
    className="ant-divider ant-divider-horizontal ant-divider-dashed"
  />
  <button
    className="ant-btn nextButton ant-btn-primary"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span>
      Next 
    </span>
    <i
      className="anticon anticon-right"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="right"
        fill="currentColor"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
        />
      </svg>
    </i>
  </button>
</div>
`;
